drivers: gpio: Fix PCA6416 configuration

kimbotto opened the pull request:
In PCA 6416 (or the compatible TCA) GPIO expander, when the interrupt pin is not connected, the DTS property _interrupt_gpios_ should not be declared. However, macro GPIO_DT_SPEC_INST_GET requires it, even if DT_INST_NODE_HAS_PROP correctly sets interrupt enable to false.

kimbotto committed:
drivers: gpio: Fix PCA6416 configuration

Fix device initialisation, allowing for empty interrupt_gpios in the
dts, as suggested by the documentation.

Signed-off-by: Mirko Bottarelli <mirko.bottarelli@gmail.com>

henrikbrixandersen commented on drivers/gpio/gpio_pca6416.c:
```diff
@@ -472,7 +472,7 @@ static DEVICE_API(gpio, api_table) = {
 		},								\
 		.interrupt_enabled = DT_INST_NODE_HAS_PROP(n, interrupt_gpios),	\
-		.gpio_int = GPIO_DT_SPEC_INST_GET(n, interrupt_gpios),	\
+		.gpio_int = GPIO_DT_SPEC_INST_GET_OR(inst, interrupt_gpios, {}),	\
```
```suggestion
		.gpio_int = GPIO_DT_SPEC_INST_GET_OR(inst, interrupt_gpios, {0}),	\
```

kimbotto committed:
drivers/gpio: Fix issue in gpio_pca6416

Added the possibility of leaving the gpio-int property empty, in case
the corresponding pin is not connected.

Signed-off-by: Mirko Bottarelli <mirko.bottarelli@gmail.com>
